lib/ansible/modules/cloud/openstack/os_group_facts.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/python
+# Copyright (c) 2016 Hewlett-Packard Enterprise Corporation
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: os_group_facts
+short_description: Retrieve facts about one or more OpenStack groups
+extends_documentation_fragment: openstack
+version_added: "2.9"
+author: "Phillipe Smith (@phsmith)"
+description:
+    - Retrieve facts about a one or more OpenStack groups
+requirements:
+    - "python >= 2.7"
+    - "openstacksdk"
+options:
+   name:
+     description:
+        - Name or ID of the group
+     required: true
+   domain:
+     description:
+        - Name or ID of the domain containing the group if the cloud supports domains
+   filters:
+     description:
+        - A dictionary of meta data to use for further filtering.  Elements of
+          this dictionary may be additional dictionaries.
+   availability_zone:
+     description:
+       - Ignored. Present for backwards compatibility
+'''
+
+EXAMPLES = '''
+# Gather facts about previously created groups
+- os_group_facts:
+    cloud: awesomecloud
```

Comment:
do these example tasks need a "register"?

Reply:
@cloudnull Sorry for the late answer.
No, this module don't need a register, It returns a variable named "openstack_groups".